### PR TITLE
Finished with Demo: Links added

### PR DESCRIPTION
**Pull request by Volminger**
Original PR:  https://github.com/KTH/devops-course/pull/732

### CODE REVIEW

**Comment by Mema5** [comment]
Hi, I'm interested to give feedback to you :)

**Comment by christinasunnegardh** [comment]
Hi, I'd like to learn more about Github Actions + CI/CD so I'd be happy to give feedback on this :)

**Comment by juliuscc** [comment]
I will give feedback

**Comment by felixlut** [comment]
I will give feedback

**Comment by christinasunnegardh** [comment]
Hi, thank you for a good demo! Here is my feedback:

- The subtitles didn't always match what the speaker was saying, this was a bit confusing to me.
- Some concepts I hadn't heard before so it became a bit hard to follow (for example "load balancer"), if you feel like you don't have time to describe it, it could be an idea to make a popup and say that there is a link in the description with info about it for example.
- Similarly in the end, it would be nice to link to some good versions of the "other demos" you are referring to :)
- I'm not sure that it is part of the criteria, but I felt a bit lost in how all of this had been set up. Maybe it would be nice to show a bit more what you pushed, what action that then was triggered etc.
- At 3:40 it's said "if all looks good here, we will go ahead..", this confuses me a bit, it would be nice to know how it should look for it to look good.

Interesting subject that got me interested in Terraform, so good job overall :)

**Comment by juliuscc** [comment]
I think the video had a good tempo. Somehow you covered a complicated issue and gave a lot of information, in a short video. Great job!

I think the topic was really interesting! You choose a clear demonstration, that highlights the benefits of terraform in a good way. For me, it became instantly clear how terraform is applicable in a real-world scenario. It was also an interesting scenario where you showed how easy it is to use three cloud providers.

I don't recognize stack path, so it threw me off. Does stack path have a super-easy way to add a load balancer or is there another reason to choosing it?

You should definitely add subtitles. With youtube, you can start with using the automatic captioning and then fix the places that the automatic captioning got wrong.

Also, the readme in your repo is a bit underwhelming.

Considering everything I think this is a great demo!

**Comment by juliuscc** [comment]
Another small addition: The first person sounded great but the second person sounded much worse. It might be difficult to fix, as you work remotely and the second person might not have the equipment but fixing that would be nice.

**Comment by felixlut** [comment]
Overall I think that the demo was informative and easy to understand.

# Positives 
* You spoke very clearly, and at no point was it hard to understand
* Nice addition of an easter egg
* Great that you've made a github repo
* I liked background part of your demo, it made it much easier to understand what's going on (i.e. the first 2 minutes)

# Negatives
* First thing first, there're 2 things I believe you should add in order to be able to pass. Nethier is big, but they should be there. 
    - Subtitles: They're there, but it seems like they're automatically generated. While they are correct most of the time, and the addition of voice over makes it really easy to understand the point you're trying to get across, I would review and correct them manually, since they are mandatory.  
    - Description: I believe a youtube description was mandatory (iirc the teachers mentioned this during last weeks lecture), with a short explanation of the problem + links and such.
* Consider adding intsructions for how to recreate what you've done in your git repo

**Comment by Mema5** [comment]
Hi! Here is my feedback. Overall: great work guys :)

## Positive points
- :+1: voice: makes it easier to follow
- :+1: good take-home message at the end

## Points to improve
- the introduction is maybe a bit long (demo start only at minute 2:55)
- maybe you could cut during infrastructure building (doable in Youtube Studio) to be below 5 min?
- I think you could demonstrate more things, I'm afraid the graders will tell you that it's not a demo but a presentation if you don't :/
- consider having a good video description with the motivation you have in the README, a link to your github repo, other links...

**Comment by pojans** [comment]
This is a really cool demo. I have set up load balancers in GKE but never a solution that balances over multiple cloud solution. 

Clear speech, nice pacing, and being a lotr fan that easter egg was really something else. 

I think the introduction is a bit long, maybe you could condense it by changing the intended audience so that you don't have to explain everything?

What you could do to take the demo to the next level is to include a slide for the take home message.

Overall, great job guys! I wish you the best of luck!

**Comment by akhilyerrapragadaa** [comment]
I'm into terraform and would like to know more about it. This video falls under my path therefore, viewed it and provided some feedback on your demo.

## Positives:
It's pristine to point.
Well described the flow and usage of each.
Nice that you made a github repo.

## Improvisations 
Consider adding some reference links (Terraform cloud docs) in youtube description. It will help the viewer just in case of additional info.

Finally, great effort and well presented.

**Comment by Volminger** [comment]
Thanks @christinasunnegardh, @juliuscc , @felixlut , @Mema5 , @pojans ,  @akhilyerrapragadaa  for the great feedback!

We agree with all of you that we need to take a look at the subtitles and fix them up. We will also try to make the introduction a bit shorter and focus more on the second part of the demo. We will possibly use your suggestion @christinasunnegardh with using popups with links to more information about the concepts we talk about. But we will try to balance this since around half of you guys seemed to like the introduction and the second half thought it was to long :relaxed:

We will also fix up the README and YouTube description which a lot of you guys suggested :blush:

@juliuscc We tried using a couple of different cloud providers for the load balancing. But the problem with the out-of-the-box load balancers that providers such as AWS and DigitalOcean have are that they only work for balancing with instances within the same provider. We then tried using CloudFlare's load balancer, but we realized that it would cost us a couple of dollars. We then read that GCP's OOTB load balancer was suppose to work with external instances. So we tried getting that to work for a while, but without any success. So we understood that we needed to put up a load balancer ourself and we decided to go with using the Container Image of [traefik](https://github.com/containous/traefik). That was when we found **Stackpath**. Set aside from the fact that we got **100$** in sign-up credits to use on the platform. The [Stackpath Terraform provider](https://www.terraform.io/docs/providers/stackpath/r/compute_workload.html) had a great way of creating and configuring containers within its instance type and we could also show that Terraform could interact well with more obscure cloud providers.
Haha that was the long explanation on why we ended up using Stackpath :laughing:

We will also try to improve the sound quality of the second part :smiley:
